### DISCUSSION TITLE
Simple DOS console emulator EMU2

### DISCUSSION THREAD

**Discussion by toncho11** [jbruchon/elks#1959] [Ideas]
https://github.com/dmsc/emu2

If this project could be compiled to ELKS it might allow some DOS text applications such as Turbo Pascal 3 and word processors and especially closed source ones to run on ELKS. 

What do you think @ghaerr ? What are the chances of porting EMU2 to ELKS?

**Comment by toncho11** [comment]
Code look like gcc with no assembler.

**Reply by ghaerr** [reply]
I'm familiar with EMU2, its a well-written and maintained CPU and DOS emulator for 32+bit systems. A quick look at the source shows it defines a 1M byte memory array for the emulated memory. Thus, there's no chance of it running on ELKS. Even if it were possible, it would be very slow since it emulates the 8086 chip (on top of the 8088 or 386, etc).

**Reply by toncho11** [reply]
Still this is currently the best chance of bringing DOS software to ELKS. Question is rather which type of speed is important. A Word processor does not make heavy calculations, but rather it needs reasonable response time on each key pressed. 

Having EMU2 on ELKS on a 486 running a text mode Word Perfect will be both impressive and usefull. A 486 is quite fast retro machine and hopefully fast enough to run EMU2.

**Reply by toncho11** [reply]
Of course it is a lot of work.
